src/olympia/addons/tests/test_commands.py

```diff
@@ -479,3 +480,80 @@ def test_correctly_ignored_because_strict_compatibility_is_enabled(self):
         apv = ApplicationsVersions.objects.get(version=addon.current_version)
         # Shouldn't have been updated to 56.* since strictCompatibilty is true.
         assert apv.max != self.firefox_56_star
+
+
+class TestDeleteAddonsNotCompatibleWithFirefoxes(TestCase):
+    def make_the_call(self):
+        call_command('process_addons',
+                     task='delete_addons_not_compatible_with_firefoxes')
+
+    def test_basic(self):
+        av_min, _ = AppVersion.objects.get_or_create(
+            application=amo.ANDROID.id, version='48.0')
+        av_max, _ = AppVersion.objects.get_or_create(
+            application=amo.ANDROID.id, version='48.*')
+        av_seamonkey_min, _ = AppVersion.objects.get_or_create(
+            application=amo.SEAMONKEY.id, version='2.49.3')
+        av_seamonkey_max, _ = AppVersion.objects.get_or_create(
+            application=amo.SEAMONKEY.id, version='2.49.*')
+        # Those add-ons should not be deleted, because they are compatible with
+        # at least Firefox or Firefox for Android.
+        addon_factory()  # A pure Firefox add-on
+        addon_factory(version_kw={'application': amo.ANDROID.id})
+        addon_with_both_firefoxes = addon_factory()
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.ANDROID.id,
+            version=addon_with_both_firefoxes.current_version,
+            min=av_min, max=av_max)
+        addon_with_thunderbird_and_android = addon_factory(
+            version_kw={'application': amo.THUNDERBIRD.id})
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.ANDROID.id,
+            version=addon_with_thunderbird_and_android.current_version,
+            min=av_min, max=av_max)
+        addon_with_firefox_and_seamonkey = addon_factory(
+            version_kw={'application': amo.FIREFOX.id})
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.SEAMONKEY.id,
+            version=addon_with_firefox_and_seamonkey.current_version,
+            min=av_seamonkey_min, max=av_seamonkey_max)
+
+        # Those add-ons should be deleted as they are only compatible with
+        # Thunderbird or Seamonkey, or both.
+        addon = addon_factory(version_kw={'application': amo.THUNDERBIRD.id})
+        addon2 = addon_factory(version_kw={'application': amo.SEAMONKEY.id,
+                                           'min_app_version': '2.49.3',
+                                           'max_app_version': '2.49.*'})
+        addon3 = addon_factory(version_kw={'application': amo.THUNDERBIRD.id})
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.SEAMONKEY.id,
+            version=addon3.current_version,
+            min=av_seamonkey_min, max=av_seamonkey_max)
+
```

Comment:
Can we throw in some addons that won't be deleted under the current filters - like non-public Thunderbird addons - and check that they arn't deleted; their `ApplicationVersions` are; and it doesn't all blow up.

Reply:
36226b917409ab0378e5f953a0841ec01f7ac319